### ISSUE TITLE
HTTP 400 (bad request) in Feeds API when expected_interactions < 0

### ISSUE THREAD

**Issue by regulartim** [intelowlproject/GreedyBear#625]
On a fresh GreedyBear instance when I opened the  "Feeds" page, I got HTTP 400. When I looked into the details of the response, it said:
`{"errors":{"expected_interactions":["Ensure this value is greater than or equal to 0."]}}`. 
In fact there was an IoC in the database with expected_interactions < 0. I don't know why the ML model came up with a negative number but we should catch this case somehow to avoid the error in the future.

**Comment by opbot-xd** [comment]
I investigated the issue where the `expected_interactions` field contained negative values, leading to HTTP 400 Bad Request errors.

It appears that the `RandomForestRegressor` model is generating negative predictions. I have confirmed this behavior by creating a reproduction script using the project's exact model hyperparameters.

I wrote this small script to reproduce the issue:

```python
import numpy as np
import pandas as pd
from sklearn.ensemble import RandomForestRegressor

# Exact hyperparameters from the GreedyBear configuration
greedybear_params = {
    "criterion": "squared_error",
    "max_depth": 11,
    "max_features": "sqrt",
    "min_samples_leaf": 3,
    "min_samples_split": 8,
    "n_estimators": 70,
    "random_state": 42
}

# 1. Root Cause: Negative Deltas in Training Data
# Simulating a scenario where an interaction count dropped (e.g., a sensor reset),
# causing a large negative target value (delta = current - previous).
data = {
    'feature_1': [10, 10, 10, 2, 2, 2, 5, 5, 5], 
    'feature_2': [5, 5, 5, 1, 1, 1, 3, 3, 3]
}
X_train = pd.DataFrame(data)

# Targets: The "10, 5" case has a large negative value representing the reset event.
y_train = pd.Series([-950, -900, -920, 20, 25, 22, 50, 55, 52]) 

model = RandomForestRegressor(**greedybear_params)
model.fit(X_train, y_train)

# Predict on a similar feature set
test_input = pd.DataFrame({'feature_1': [10], 'feature_2': [5]})
prediction = model.predict(test_input)[0]

print(f"Raw Prediction using GreedyBear Config: {prediction:.2f}")
```

<img width="487" height="31" alt="Image" src="https://github.com/user-attachments/assets/9ecc7eba-8f8f-464d-8da8-d640d24031e2" />

#### Physical Interpretation
While "negative attacks" are physically impossible, the model generates these values due to the way training targets are calculated:
1.  **Metric Calculation**: The model targets the **change** in interactions (`Today - Yesterday`).
2.  **Sensor Anomalies**: If a honeypot sensor resets its internal counter (e.g., from 1,000 to 0), the calculated change is negative (e.g., -1,000).
3.  **Model Inference**: The regression model learns these negative patterns. When it encounters similar feature vectors in the future, it predicts a negative expected change.

Since the API correctly validates that `expected_interactions` must be non-negative, these unhandled predictions resulted in validation errors.

#### Resolution
I can implement a fix to clip the model's output at 0 using `prediction = max(prediction, 0)`. This ensures that "negative expectations" are treated as "zero expected new interactions," satisfying the API's constraints while maintaining system stability.

**Comment by regulartim** [comment]
Interesting. That might be the reason. Your suggested resolution seems appropriate, feel free to implement it. I will assign the issue to you.
